app/descriptor/views.py

```diff
@@ -29,50 +34,163 @@ def new_descriptor():
             if v is not None and len(v) != 0:
                 values.append(v)
         descriptor = Descriptor(name=form.name.data, values=values)
+        if Descriptor.query.filter(Descriptor.name == form.name.data).first() \
+                is not None:
+            flash('Descriptor {} already exists.'.format(descriptor.name),
+                  'form-error')
+        else:
+            db.session.add(descriptor)
+            db.session.commit()
+            flash('Descriptor {} successfully created'.format(descriptor.name),
+                  'form-success')
+            return redirect(url_for('descriptor.new_descriptor'))
+    return render_template('descriptor/new_descriptor.html', form=form)
+
+
+@descriptor.route('/<int:desc_id>', methods=['GET', 'POST'])
+@login_required
+def descriptor_info(desc_id):
+    """Display the descriptor info."""
+    descriptor = Descriptor.query.get(desc_id)
+    is_option = len(descriptor.values) != 0
+    if descriptor is None:
+        abort(404)
+    return render_template('descriptor/manage_descriptor.html',
+                           desc=descriptor, is_option=is_option)
+
+
+@descriptor.route('/<int:desc_id>/name', methods=['GET', 'POST'])
+@login_required
+def edit_name(desc_id):
+    """Edit a descriptor's name."""
+    descriptor = Descriptor.query.get(desc_id)
+    is_option = len(descriptor.values) != 0
+    old_name = descriptor.name
+    if descriptor is None:
```

Comment:
Does this check need to be earlier? I think likes 67 and 68 will fail if `descriptor` is `None`.